readthedocs/notifications/messages.py

```diff
@@ -0,0 +1,434 @@
+
+from django.utils.translation import gettext_noop as _
+
+from readthedocs.doc_builder.exceptions import (
+    BuildAppError,
+    BuildCancelled,
+    BuildMaxConcurrencyError,
+    BuildUserError,
+    MkDocsYAMLParseError,
+    UnsupportedSymlinkFileError,
+)
+from readthedocs.projects.constants import BUILD_COMMANDS_OUTPUT_PATH_HTML
+
+from .constants import ERROR, INFO, NOTE, SOLID, TIP, WARNING
+
+
+class Message:
+    def __init__(self, id, header, body, type, icon=None, icon_style=SOLID):
+        self.id = id
+        self.header = header
+        self.body = body
+        self.type = type  # (ERROR, WARNING, INFO, NOTE, TIP)
+        self.icon = icon
+        self.icon_style = icon_style  # (SOLID, DUOTONE)
+        self.format_values = {}
+
+    def __repr__(self):
+        return f"<Message: {self.id}>"
+
+    def __str__(self):
+        return f"Message: {self.id} | {self.header}"
+
+    def set_format_values(self, format_values):
+        self.format_values = format_values or {}
+
+    def get_display_icon(self):
+        if self.icon:
+            return self.icon
+
+        if self.type == ERROR:
+            return "fa-exclamation"
+        if self.type == WARNING:
+            return "fa-triangle-exclamation"
+        if self.type == INFO:
+            return "fa-????"
+        if self.type == NOTE:
+            return "fa-????"
+        if self.type == TIP:
+            return "fa-????"
+
+    def get_rendered_header(self):
+        return self.header.format(**self.format_values)
+
+    def get_rendered_body(self):
+        return self.body.format(**self.format_values)
+
+
+CONFIG_YAML_MESSAGES = []
+
+# TODO: review the copy of these notifications/messages on PR review and adapt them.
+# Most of them are copied from what we had in `readthedocs.doc_builder.exceptions`
+# and slightly adapted to have a header and a better body.
+BUILD_MESSAGES = [
+    Message(
+        id=BuildAppError.GENERIC_WITH_BUILD_ID,
```

Comment:
I was going to suggest this pattern on some of the other messages. It seems wise to reference a shared constant instead of a string here, or I could see the lookup ids going out of sync.

Reply:
Yeah, this is the correct pattern.

The places where I was using hard-coded strings are WIP and I plan to move them to this pattern. The IDs may be under a class defining an exception or just global constants.